### PR TITLE
PLANNER-1182 & PLANNER-1183: Improvements to Avaliability and Shift Rosters

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**

### CODE REVIEW

**Comment by Christopher-Chianelli** [comment]
jenkins retest this

File: employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/availabilityroster/AvailabilityGridObjectPopup.java [hunk 1]
```diff
@@ -107,12 +124,38 @@ public void init(final AvailabilityGridObject availabilityGridObject) {
...
+        setup(availabilityView);
+
+        popupFactory.getFormPopup(this).ifPresent((fp) -> {
+            formPopup = fp;
+            formPopup.showFor(availabilityGridObject);
+        });
+    }
+
+    public void createNewAvailability() {
+        setup(new EmployeeAvailabilityView(tenantStore.getCurrentTenantId(), new Employee(), LocalDate.now().atTime(0, 0), LocalDate.now().plusDays(1).atTime(0, 0), EmployeeAvailabilityState.UNAVAILABLE));
```
Comment (yurloc) [inline employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/availabilityroster/AvailabilityGridObjectPopup.java]: Using new lines would really improve code readability here.

File: employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObjectPopup.java [hunk 2]
```diff
@@ -111,27 +119,52 @@ public void init(final ShiftGridObject shiftGridObject) {
...
+    }
+
+    private void setup(ShiftView shiftView) {
         employeeSelect.clear();
         employeeSelect.addItem(translationService.format(OptaWebUIConstants.Shift_unassigned), "-1");
 
         SpotRestServiceBuilder.getSpotList(tenantStore.getCurrentTenantId(), FailureShownRestCallback.onSuccess(spots -> {
             spots.forEach(s -> this.spotSelect.addItem(s.getName(), s.getId().toString()));
-            spotSelect.setSelectedIndex(spots.indexOf(shiftGridObject.getSpot()));
+            if (null == shiftView.getSpotId()) {
```
Comment (yurloc) [inline employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObjectPopup.java]: No need for [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) in Java.

File: employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObjectPopup.java [hunk 3]
```diff
@@ -111,27 +119,52 @@ public void init(final ShiftGridObject shiftGridObject) {
...
+                spotSelect.setSelectedIndex(0);
+            } else {
+                spotSelect.setSelectedIndex(spots.indexOf(shiftGridObject.getSpot()));
+            }
         }));
 
         EmployeeRestServiceBuilder.getEmployeeList(tenantStore.getCurrentTenantId(), FailureShownRestCallback.onSuccess(employees -> {
             employees.forEach(e -> employeeSelect.addItem(e.getName(), e.getId().toString()));
-            employeeSelect.setSelectedIndex((shiftGridObject.getEmployee() == null) ? 0 : employees.indexOf(shiftGridObject.getEmployee()) + 1);
+            if (null == shiftView.getEmployeeId()) {
```
Comment (yurloc) [inline employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObjectPopup.java]: No need for [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) in Java.

**Comment by Christopher-Chianelli** [comment]
Merged, fixed merge issue and adopted @yurloc suggested code changes

**Pull request merged**